apps/server/src/modules/task/controller/api-test/task.api.spec.ts

```diff
@@ -1456,4 +1459,218 @@ describe('Task Controller (API)', () => {
+				student2.account,
+				student2.user,
+				student3.account,
+				student3.user,
+				course1,
+				course2,
+				course1Task1,
+				course1Task2,
+				course1Task3,
+				course1Task4,
+				course2Task1,
+				course2Task2,
+				course2Task3,
+			]);
+			em.clear();
+
+			return {
+				teacher,
+				student1,
+				student2,
+				student3,
+				course1,
+				course2,
+				course1Task1,
+				course1Task2,
+				course1Task3,
+				course1Task4,
+				course2Task1,
+				course2Task2,
+				course2Task3,
+			};
+		};
+
+		beforeAll(async () => {
+			const module: TestingModule = await Test.createTestingModule({
+				imports: [ServerTestModule],
+			}).compile();
+
+			app = module.createNestApplication();
+			await app.init();
+			em = module.get(EntityManager);
+			apiRequest = new TestRequest(app, 'tasks');
+		});
+
+		afterAll(async () => {
+			await app.close();
+		});
+
+		beforeEach(async () => {
+			await cleanupCollections(em);
+		});
+
+		describe('when task have assigment', () => {
+			it('finds tasks to which student is assigned', async () => {
+				const { student1, course2Task1 } = await setup();
+
+				const response = await apiRequest.get(undefined, student1.account);
+
+				const { total, data } = response.body as TaskListResponse;
+				expect(response.statusCode).toBe(200);
+				const taskIds = data.map((task) => task.id);
+				expect(taskIds).toContain(course2Task1.id);
+				expect(total).toBe(3);
+			});
+
+			it('find finished tasks to which student is assigned ', async () => {
+				const { student2, course1Task1, course2Task2 } = await setup();
+
+				const response = await apiRequest.get('/finished', student2.account);
+
+				const { total, data } = response.body as TaskListResponse;
+				expect(response.statusCode).toBe(200);
+				const taskIds = data.map((task) => task.id);
+				expect(taskIds).toContain(course1Task1.id);
+				expect(taskIds).toContain(course2Task2.id);
+				expect(total).toBe(2);
+			});
+
+			it('student does not find tasks to which he is assigned, if he does not belong to course', async () => {
+				const { student3, course1Task1 } = await setup();
+
+				const response = await apiRequest.get(undefined, student3.account);
+
+				const { data } = response.body as TaskListResponse;
+				expect(response.statusCode).toBe(200);
+				const taskIds = data.map((task) => task.id);
+				expect(taskIds).not.toContain(course1Task1.id);
+			});
+
+			it('student does not find tasks, it task has users assigned, but himself is not assigned', async () => {
+				const { student1, course1Task2 } = await setup();
+
+				const response = await apiRequest.get(undefined, student1.account);
+
+				const { data } = response.body as TaskListResponse;
+				expect(response.statusCode).toBe(200);
+				const taskIds = data.map((task) => task.id);
+				expect(taskIds).not.toContain(course1Task2.id);
+			});
+			it('student does not find finished tasks, it tasks have users assigned, but himself is not assigned', async () => {
```

Comment:
we specifically want to test for some edge cases, which could be created from inconsistency of data
I want to have as clear as possible description - must be said that the task has some users. Because not being assigned, might mean nobody is assigned, and that is something else.